lib/rubocop/cop/style/safe_navigation.rb

```diff
@@ -51,18 +61,18 @@ class SafeNavigation < Cop
               (send $_ {:nil? :!}) nil
               {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
             ...)
-            (and
-              {(send (send $_ :nil?) :!) $_}
-              {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
-            ...)
-            (or
-              (send $_ {:nil? :!})
-              {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
-            ...)
           }
         PATTERN
 
+        def_node_matcher :candidate_that_may_introduce_nil, <<-PATTERN
+          (and
+            {(send (send $_ :nil?) :!) $_}
+            {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
+          ...)
+        PATTERN
+
         def on_if(node)
+          return unless modifier_if?(node)
```

Comment:
Can we move the check to `auto_correct` method?
If move, the cop can detect non modifier if, and the cop doesn't apply auto-correction.